pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -244,59 +248,63 @@ func ensureTeamUsersAreInSync(workflowCtx *workflow.Context, teamID string, team
+	_, _, err = workflowCtx.SdkClient.TeamsApi.AddTeamUser(workflowCtx.Context, workflowCtx.OrgID, teamID, &toAdd).Execute()
 	if err != nil {
 		return workflow.Terminate(workflow.TeamUsersNotReady, err.Error())
 	}
 
 	return workflow.OK()
 }
 
-func fetchTeamByID(workflowCtx *workflow.Context, teamID string) (*mongodbatlas.Team, error) {
+func fetchTeamByID(workflowCtx *workflow.Context, teamID string) (*admin.Team, error) {
 	workflowCtx.Log.Debugf("fetching team %s from atlas", teamID)
-	atlasTeam, _, err := workflowCtx.Client.Teams.Get(workflowCtx.Context, workflowCtx.OrgID, teamID)
+	atlasTeamResponse, _, err := workflowCtx.SdkClient.TeamsApi.GetTeamById(workflowCtx.Context, workflowCtx.OrgID, teamID).Execute()
 	if err != nil {
 		return nil, err
 	}
+	atlasTeam := &admin.Team{Id: atlasTeamResponse.Id, Name: *atlasTeamResponse.Name, Links: atlasTeamResponse.Links}
 
 	return atlasTeam, nil
 }
 
-func fetchTeamByName(workflowCtx *workflow.Context, teamName string) (*mongodbatlas.Team, error) {
+func fetchTeamByName(workflowCtx *workflow.Context, teamName string) (*admin.Team, error) {
 	workflowCtx.Log.Debugf("fetching team named %s from atlas", teamName)
-	atlasTeam, resp, err := workflowCtx.Client.Teams.GetOneTeamByName(workflowCtx.Context, workflowCtx.OrgID, teamName)
+	atlasTeamResponse, resp, err := workflowCtx.SdkClient.TeamsApi.GetTeamByName(workflowCtx.Context, workflowCtx.OrgID, teamName).Execute()
 	if err != nil {
 		if resp.StatusCode == http.StatusNotFound {
 			return nil, nil
 		}
 
 		return nil, err
 	}
+	atlasTeam := &admin.Team{Id: atlasTeamResponse.Id, Name: *atlasTeamResponse.Name, Links: atlasTeamResponse.Links}
 
 	return atlasTeam, nil
 }
 
-func createTeam(workflowCtx *workflow.Context, atlasTeam *mongodbatlas.Team) (*mongodbatlas.Team, error) {
+func createTeam(workflowCtx *workflow.Context, atlasTeam *admin.Team) (*admin.Team, error) {
 	workflowCtx.Log.Debugf("create team named %s in atlas", atlasTeam.Name)
-	atlasTeam, _, err := workflowCtx.Client.Teams.Create(workflowCtx.Context, workflowCtx.OrgID, atlasTeam)
+	atlasTeam, _, err := workflowCtx.SdkClient.TeamsApi.CreateTeam(workflowCtx.Context, workflowCtx.OrgID, atlasTeam).Execute()
 	if err != nil {
 		return nil, err
 	}
 
 	return atlasTeam, nil
 }
 
-func renameTeam(workflowCtx *workflow.Context, atlasTeam *mongodbatlas.Team, newName string) (*mongodbatlas.Team, error) {
+func renameTeam(workflowCtx *workflow.Context, atlasTeam *admin.Team, newName string) (*admin.Team, error) {
 	if atlasTeam.Name == newName {
 		return atlasTeam, nil
 	}
 
-	workflowCtx.Log.Debugf("updating name of team %s in atlas", atlasTeam.ID)
-	atlasTeam, _, err := workflowCtx.Client.Teams.Rename(workflowCtx.Context, workflowCtx.OrgID, atlasTeam.ID, newName)
+	workflowCtx.Log.Debugf("updating name of team %s in atlas", atlasTeam.Id)
+	teamUpdate := admin.TeamUpdate{Links: atlasTeam.Links, Name: newName}
```

Comment:
```suggestion
	teamUpdate := admin.TeamUpdate{Name: newName}
```

Setting the `Links` property in the `TeamUpdate` struct is not desired. As per godoc this is a read-only field:
```
type TeamUpdate struct {
	// List of one or more Uniform Resource Locators (URLs) that point to API sub-resources, related API resources, or both. RFC 5988 outlines these relationships.
	// Read only field.
	Links *[]Link `json:"links,omitempty"`
...
}
```